File: src/main/kotlin/dev/jefy/connectpro/user/applicaion/dtos/UserData.java
package dev.jefy.connectpro.user.applicaion.dtos;


import dev.jefy.connectpro.user.domain.model.AuthUser;
import dev.jefy.connectpro.user.domain.model.User;
import dev.jefy.connectpro.user.domain.vo.Email;
import dev.jefy.connectpro.user.domain.vo.UserId;
import dev.jefy.connectpro.user.domain.vo.UserRole;

/**
 * @author Jôph Yamba
 */
public record UserData (UserId id, Email email, String name, UserRole role, String imageUrl) {
    public static UserData fromDomain(User user) {
        return new  UserData(
                user.getId(),
                user.getEmail(),
                user.getName(),
                user.getRole(),
                user.getProfileImage().value()
        );
    }
        public static UserData fromAuth(AuthUser user) {
        return new  UserData(
                UserId.of(user.id()),
                Email.of(user.email()),
                user.name(),
                user.role(),
                user.imageUrl()
        );
    }
    
    
}

File: src/main/kotlin/dev/jefy/connectpro/user/application/dtos/UserData.kt
package dev.jefy.connectpro.user.application.dtos

import dev.jefy.connectpro.shared.infrastructure.file_storage.ImageUrlResolver
import dev.jefy.connectpro.user.domain.model.AuthUser
import dev.jefy.connectpro.user.domain.model.User
import java.util.*

data class UserData(
    val id: UUID,
    val email: String,
    val name: String,
    val image: String? = null
)

fun User.toUserData(resolver: ImageUrlResolver): UserData = UserData(
    id = this.id.value,
    email = this.email.value,
    name = this.name,
    image = resolver.resolve(this.image)
)

fun AuthUser.toUserData(): UserData = UserData(
    id = this.id,
    email = this.email,
    name = this.fullname,
    image = this.image
)